File: Data Structures & Algorithms/find-duplicate-integer/submission-4.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        f = 0
        s = 0

        while True:
            f = nums[nums[f]]
            s = nums[s]
            if nums[f] == nums[s]:
                print(f, s)  
                break
        s2 = 0
        while True:
            s2 = nums[s2]
            s = nums[s]
            if s2 == s:
                break
        return s  

File: Data Structures & Algorithms/majority-element/submission-4.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    majorityElement(nums) {
        const counts = new Map();
        const uniqueNums = new Set();
        let res = nums[0];
        for(let n of nums) {
            uniqueNums.add(n);
            counts.set(n, (counts.get(n) || 0) + 1);
        }

        for(let n of uniqueNums) {
            if(counts.get(n) > (nums.length/2))
                res = n;
        }

        return res;

    }
}
